Project-Spaced-Repetition-Tracker/storage.js

```diff
@@ -0,0 +1,50 @@
+// This is a scaffolding file we have provided for you which allows you to manage stored data for your application.
+// It can be loaded into index.html.
+// You should not need to modify it to complete the project.
+
+/**
+ * Get a list of user ids
+ *
+ * @returns {string[]} List of user id strings
+ */
+export function getUserIds() {
+  return ["1", "2", "3", "4", "5"];
+}
+
+/**
+ * Get data associated with a specific user.
+ *
+ * @param {string} userId The user id to get data for
+ * @returns {any[] | null} The data associated with the user
+ */
+export function getData(userId) {
+  try {
+    return JSON.parse(localStorage.getItem(`stored-data-user-${userId}`));
+  } catch (_) {
+    return null;
+  }
+}
+
+/**
+ * Store data for a specific user.
+ *
+ * @param {string} userId The user id to store data for
+ * @param {any[]} data The data to store
+ */
+export function setData(userId, data) {
```

Comment:
```suggestion
 * Store data for a specific user. If there was already data for this user, this function preserves it and adds the new data at the end.
 *
 * @param {string} userId The user id to store data for
 * @param {any[]} data The data to store
 */
export function addData(userId, data) {
```

That said, I think it's also valid for us to just provide get/set and require people to concat themselves... I can imagine some of the group extensions requiring deleting existing data, at which point `clear` isn't just a convenience...

Reply:
Thanks, I think that does indeed better express the intention.

I did also consider the getter/setter idea, but I was aiming for more of a database-like API, with a pseudo data-structure.